File: sdk/typescript/src/tokens.ts
/** LLM token counting via tiktoken-rs in the Rust core. */

import {
  configureTokenizerDefaultsNative,
  countJsonTokensNative,
  countTokensBatchNative,
  countTokensNative,
} from "./native.js";

export function countTokens(text: string): number {
  return countTokensNative(text);
}

export function countTokensBatch(texts: string[]): number[] {
  return countTokensBatchNative(texts);
}

export function countJsonTokens(value: unknown): number {
  return countJsonTokensNative(value);
}

export function configureTokenizerDefaults(options?: {
  encoding?: string;
  allowedSpecial?: "all" | "none";
}): void {
  configureTokenizerDefaultsNative(options?.encoding, options?.allowedSpecial);
}

File: sdk/go/tokens.go
package cytindexer

// CountTokens returns the tiktoken count for text under the configured encoding.
func CountTokens(text string) (int64, error) {
	return cgoCountTokens(text)
}

// CountJsonTokens returns the token count for compact JSON text.
func CountJsonTokens(jsonStr string) (int64, error) {
	return cgoCountJsonTokens(jsonStr)
}

// ConfigureTokenizerDefaults overrides tokenizer defaults in the native core.
// Pass empty configJSON to leave current settings unchanged.
func ConfigureTokenizerDefaults(configJSON string) error {
	return cgoConfigureTokenizerDefaults(configJSON)
}
